ProjectManagmentSystem/src/Modules/ProjectsModule/components/ProjectsList/ProjectsList.tsx

```diff
@@ -9,46 +9,86 @@ import Modal from "react-bootstrap/Modal";
 import DeleteData from "./../../../SharedModule/components/DeleteData/DeleteData";
 import { useToast } from "../../../Context/ToastContext";
 import Loading from "../../../SharedModule/components/Loading/Loading";
+
+
+
+
 export default function ProjectsList() {
   const { requestHeaders, baseUrl }: any = useContext(AuthContext);
   const { getToast } = useToast();
   const [projectsList, setProjectsList] = useState([]);
   const [ProjectId, setProjectId] = useState(0);
   const [isLoading, setIsLoading] = useState(false);
-
+  const [totalResults, setTotalResults] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pageSize, setPageSize] = useState(5); 
+  const [title, setTitle] = useState('');
+  const [arrayOfPages, setArrayOfPages] = useState<number[]>([]);
   const [showDelete, setShowDelete] = useState(false);
-
+  const [transitionClass, setTransitionClass] = useState('');
+  
   const handleDeleteClose = () => setShowDelete(false);
   const handleDeleteShow = (id: number) => {
     setProjectId(id);
     setShowDelete(true);
   };
   const navigate = useNavigate();
 
-  ////API's
+  
+
 
-  //Get ALL Projects API
-  const getProjectsList = async () => {
+
+  //*=========================================================>> Get ALL Projects API
+  const getProjectsList = async (title='', pageSize=5, pageNumber=1  ) => {
     try {
-      let response = await axios.get(`${baseUrl}/Project/manager`, {
+      let response = await axios.get(`${baseUrl}/Project/manager?pageSize=${pageSize}&pageNumber=${pageNumber}`, {
```

Comment:
We can add the pageSize and pageNumber in the params object  as well.

Reply:
done